src/core/model.js

```diff
@@ -274,9 +277,15 @@ export default class Model extends Object3D {
     // application, these are marked deprecated in 5.0, remove them in deck.gl in 6.0.
     // Disabling since it gets too noisy in console, these are documented as deprecated.
     // this._checkForDeprecatedUniforms(uniforms);
-    checkUniformValues(uniforms, this.id);
-    Object.assign(this.uniforms, uniforms);
-    this.setNeedsRedraw();
+    const somethingChanged = Object.keys(uniforms).some(key => {
+      return !equals(uniforms[key], this.uniforms[key]);
```

Comment:
Some uniforms can be instances of `Texture`s and `Sampler`s. As long as they don't crash `equals`...

Reply:
Nit: Use a (for in) to avoid minting a keys array?